src/display/annotation_layer.js

```diff
@@ -2508,15 +2510,36 @@ class AnnotationLayer {
         }
         if (Array.isArray(rendered)) {
           for (const renderedElement of rendered) {
+            renderedElement.style.zIndex = zIndex++;
+            const contentElement =
+              renderedElement.firstChild || renderedElement;
+            contentElement.id = `${AnnotationPrefix}${data.id}`;
+
             div.append(renderedElement);
+            accessibilityManager?.moveElementInDOM(
+              div,
+              renderedElement,
+              contentElement,
+              /* isRemovable = */ false
+            );
```

Comment:
I wonder we perhaps should add a (private) helper method for this code, to reduce the risk of future bugs, rather than effectively repeating this twice?